src/Forms/Validation.jsx

```diff
@@ -0,0 +1,83 @@
+import React, { createContext, Component } from "react";
+import Schema from "async-validator";
+import PropTypes from "prop-types";
+
+const DEFAULT_STATE = {
+  errors: {}
+};
+
+export const ValidationContext = createContext(DEFAULT_STATE);
+
+export default class Validation extends Component {
+  state = DEFAULT_STATE;
+
+  valueOfTarget = target => {
+    const { type, checked, value, id } = target;
+    switch (type) {
+      case "checkbox":
+        return { [id]: checked };
+      default:
+        return { [id]: value };
+    }
+  };
+
+  validateForm = e => {
+    if (e && e.preventDefault) {
+      e.preventDefault();
+    }
+
+    const { rules } = this.props;
+    const fields = Array.from(e.target.elements)
```

Comment:
is e.target.elements already an array?

Reply:
no its not already an array. why do you ask?

Reply:
Thought it was an array

Reply:
ehm, tbh this piece of code was there. I will double check.